packages/test/src/integration-tests.ts

```diff
@@ -1121,4 +1122,64 @@ export function runIntegrationTests(codec?: PayloadCodec): void {
     });
     t.pass();
   });
+
+  if ('promiseHooks' in v8) {
+    // Skip in old node versions
+    test('Stack trace query returns stack that makes sense', async (t) => {
+      const { client } = t.context;
+      const workflowId = uuid4();
+      const rawStacks = await client.execute(workflows.stackTracer, {
+        taskQueue: 'test',
+        workflowId,
+      });
+      const [stack1, stack2] = rawStacks.map((r) =>
+        r
+          .split('\n\n')
+          .map((s) => cleanOptionalStackTrace(`\n${s}`))
+          .join('\n')
+      );
+      t.is(
+        stack1,
+        `
+    at new Promise (<anonymous>)
+    at scheduleActivityNextHandler (workflow/src/workflow.ts)
+    at scheduleActivity (workflow/src/workflow.ts)
+    at activityProxyFunction (workflow/src/workflow.ts)
+    at stackTracer (test/src/workflows/stack-tracer.ts)
+
+    at executeChild (workflow/src/workflow.ts)
```

Comment:
Is each separate "paragraph" a separate "coroutine"/wait?

Reply:
Arguably indention could help with this. Maybe in another issue we can add a description for each promise we create, but in the meantime, something like:

```
Waiting on Promise:
    at new Promise (<anonymous>)
    at scheduleActivityNextHandler (workflow/src/workflow.ts)
    at scheduleActivity (workflow/src/workflow.ts)
    at activityProxyFunction (workflow/src/workflow.ts)
    at stackTracer (test/src/workflows/stack-tracer.ts)

Waiting on Promise.race:
    at Function.race (<anonymous>)
    at stackTracer (test/src/workflows/stack-tracer.ts)
```

Could be clearer

Reply:
That's a good idea.
Note that I decided to ignore promises waited on by `race` and friends since you reviewed this so the output should be cleaner.